compute_sdk/globus_compute_sdk/sdk/login_manager/authorizer_login_manager.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+import logging
+
+import globus_sdk
+from globus_compute_sdk.sdk.login_manager.protocol import LoginManagerProtocol
+from globus_compute_sdk.sdk.web_client import WebClient
+from globus_sdk.scopes import AuthScopes
+
+from .manager import ComputeScopeBuilder
+
+log = logging.getLogger(__name__)
+
+ComputeScopes = ComputeScopeBuilder()
+
+
+class AuthorizerLoginManager(LoginManagerProtocol):
+    """
+    Implements a LoginManager that can be instantiated with authorizers.
+    This manager can be used to create an Executor with authorizers created
+    from previously acquired tokens, rather than requiring a Native App login
+    flow or Client credentials.
+    """
+
+    def __init__(self, authorizers: dict[str, globus_sdk.RefreshTokenAuthorizer]):
+        self.authorizers = authorizers
+
+    def get_auth_client(self) -> globus_sdk.AuthClient:
+        return globus_sdk.AuthClient(authorizer=self.authorizers[AuthScopes.openid])
+
+    def get_web_client(
+        self, *, base_url: str | None = None, app_name: str | None = None
+    ) -> WebClient:
+        return WebClient(
+            base_url=base_url,
+            app_name=app_name,
+            authorizer=self.authorizers[ComputeScopes.resource_server],
+        )
+
+    def ensure_logged_in(self):
+        return True
```

Comment:
Discussed verbally and in Slack: in the context where we expect the `AuthorizerLoginManager` will be utilized, the token in use here has already been setup&nbsp;&mdash;&nbsp;the typical case would be a service wrapping our SDK, like a Globus Action Provider, that has received the token as part of the API request.  In that scenario, then, this method does not necessarily need to do anything, so `return True` or `pass` would be acceptable approaches.

However, in looking at the _intent_ of this method in sister classes, it appears to be playing double-duty to both ensure that the token has access to [all of the scopes](https://github.com/funcx-faas/funcX/blob/main/compute_sdk/globus_compute_sdk/sdk/login_manager/manager.py#L45-L63) and initiating a login-flow if not.  Clearly, the login-flow is not appropriate here, but we wondered what _might_ be?

Motivated by a desire to "do the right thing" in the non-happy-path case, we landed on seeing if it's feasible to do the same thing, but just break/fail instead of trying to initiate a login-flow.  I expect this won't be an issue 99% of the time, but looking to be professional in the 1% case where something _is_ awry.

Reply:
Good chat over this. I've updated `ensure_logged_in` to check whether authorizers for each of the required scopes are present. Note, it does not check the validity of the authorizers, but does ensure they exist. I've also updated the example to invoke this.